# Data Exploration

In [25]:
import pandas as pd

## Preprocessing on our data

In [26]:
df = pd.read_csv("../data/data.csv")
df.head(5)

Country ISO3 Country Name  Indicator Id  \
0          AFG  Afghanistan         27959   
1          AFG  Afghanistan         27960   
2          AFG  Afghanistan         27961   
3          AFG  Afghanistan         27962   
4          AFG  Afghanistan         28158   

                                           Indicator Subindicator Type  2006  \
0                    Overall Global Gender Gap Index             Index   NaN   
1   Global Gender Gap Political Empowerment subindex              Rank   NaN   
2   Global Gender Gap Political Empowerment subindex             Index   NaN   
3                    Overall Global Gender Gap Index              Rank   NaN   
4  Global Gender Gap Economic Participation and O...              Rank   NaN   

   2007  2008  2009  2010  2011  2012  2013  2014  2015  2016  2017  2018  \
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   

   2019     2020  
0   NaN    0.444  
1   NaN  111.000  
2   NaN    0.132  
3   NaN  156.000  
4   NaN  156.000

We would like to clean up our data by dropping rows with NaN

In [27]:
# drop row with NaN
i=2006
while i<=2020:
    df.dropna(subset = [str(i)], inplace=True)
    i+=1

In [28]:
df.head(5)

Country ISO3 Country Name  Indicator Id  \
48          ALB      Albania         27959   
49          ALB      Albania         27960   
50          ALB      Albania         27961   
51          ALB      Albania         27962   
52          ALB      Albania         28158   

                                            Indicator Subindicator Type  \
48                    Overall Global Gender Gap Index             Index   
49   Global Gender Gap Political Empowerment subindex              Rank   
50   Global Gender Gap Political Empowerment subindex             Index   
51                    Overall Global Gender Gap Index              Rank   
52  Global Gender Gap Economic Participation and O...              Rank   

        2006      2007      2008      2009     2010     2011      2012  \
48    0.6607    0.6685    0.6591    0.6601   0.6726   0.6748    0.6655   
49  105.0000  117.0000  119.0000  125.0000  97.0000  99.0000  105.0000   
50    0.0375    0.0375    0.0413    0.0413   0.0785   0.0784    0.0753   
51   61.0000   66.0000   87.0000   91.0000  78.0000  78.0000   91.0000   
52   38.0000   36.0000   62.0000   63.0000  51.0000  38.0000   63.0000   

        2013     2014    2015    2016    2017    2018    2019    2020  
48    0.6412   0.6869   0.701   0.704   0.728   0.734   0.769   0.770  
49  130.0000  55.0000  50.000  53.000  31.000  34.000  23.000  30.000  
50    0.0256   0.1834   0.214   0.214   0.284   0.284   0.376   0.377  
51  108.0000  83.0000  70.000  62.000  38.000  34.000  20.000  25.000  
52   87.0000  78.0000  69.000  73.000  70.000  54.000  36.000  35.000

Exporting 3 dataframe, data with no NAN, data with only Index, data with only Rank

In [29]:
df1 = df[df['Subindicator Type'] == 'Index']
df2 = df[df['Subindicator Type'] == 'Rank']

In [30]:
# show data contains only index
df1.head(5)

Country ISO3 Country Name  Indicator Id  \
48          ALB      Albania         27959   
50          ALB      Albania         27961   
53          ALB      Albania         28159   
55          ALB      Albania         28161   
57          ALB      Albania         28163   

                                            Indicator Subindicator Type  \
48                    Overall Global Gender Gap Index             Index   
50   Global Gender Gap Political Empowerment subindex             Index   
53  Global Gender Gap Economic Participation and O...             Index   
55  Global Gender Gap Educational Attainment Subindex             Index   
57     Global Gender Gap Health and Survival Subindex             Index   

      2006    2007    2008    2009    2010    2011    2012    2013    2014  \
48  0.6607  0.6685  0.6591  0.6601  0.6726  0.6748  0.6655  0.6412  0.6869   
50  0.0375  0.0375  0.0413  0.0413  0.0785  0.0784  0.0753  0.0256  0.1834   
53  0.6613  0.6885  0.6491  0.6532  0.6808  0.7129  0.6663  0.6324  0.6534   
55  0.9888  0.9925  0.9907  0.9906  0.9942  0.9809  0.9890  0.9755  0.9700   
57  0.9553  0.9553  0.9553  0.9553  0.9370  0.9268  0.9313  0.9313  0.9409   

     2015   2016   2017   2018   2019   2020  
48  0.701  0.704  0.728  0.734  0.769  0.770  
50  0.214  0.214  0.284  0.284  0.376  0.377  
53  0.671  0.668  0.673  0.701  0.743  0.748  
55  0.972  0.986  0.986  0.987  0.999  0.999  
57  0.947  0.947  0.968  0.963  0.958  0.980

In [31]:
# show data contains only rank
df2.head(5)

Country ISO3 Country Name  Indicator Id  \
49          ALB      Albania         27960   
51          ALB      Albania         27962   
52          ALB      Albania         28158   
54          ALB      Albania         28160   
56          ALB      Albania         28162   

                                            Indicator Subindicator Type  \
49   Global Gender Gap Political Empowerment subindex              Rank   
51                    Overall Global Gender Gap Index              Rank   
52  Global Gender Gap Economic Participation and O...              Rank   
54  Global Gender Gap Educational Attainment Subindex              Rank   
56     Global Gender Gap Health and Survival Subindex              Rank   

     2006   2007   2008   2009   2010   2011   2012   2013   2014   2015  \
49  105.0  117.0  119.0  125.0   97.0   99.0  105.0  130.0   55.0   50.0   
51   61.0   66.0   87.0   91.0   78.0   78.0   91.0  108.0   83.0   70.0   
52   38.0   36.0   62.0   63.0   51.0   38.0   63.0   87.0   78.0   69.0   
54   57.0   48.0   68.0   73.0   52.0   87.0   76.0   92.0   98.0   98.0   
56  110.0  116.0  118.0  122.0  131.0  135.0  133.0  134.0  139.0  142.0   

     2016   2017   2018   2019   2020  
49   53.0   31.0   34.0   23.0   30.0  
51   62.0   38.0   34.0   20.0   25.0  
52   73.0   70.0   54.0   36.0   35.0  
54   88.0   87.0   91.0   40.0   42.0  
56  141.0  120.0  134.0  145.0  147.0

In [32]:
# export data
df.to_csv (r'../data/data_clean.csv', index = False, header=True)
df1.to_csv (r'../data/data_index.csv', index = False, header=True)
df2.to_csv (r'../data/data_rank.csv', index = False, header=True)

## Diving deep into data

In [33]:
uniq = df['Indicator'].unique()
print("There are "+ str(len(uniq)) + " indicators in our data: ")
print(uniq)

There are 5 indicators in our data: 
['Overall Global Gender Gap Index'
 'Global Gender Gap Political Empowerment subindex'
 'Global Gender Gap Economic Participation and Opportunity Subindex'
 'Global Gender Gap Educational Attainment Subindex'
 'Global Gender Gap Health and Survival Subindex']


In [35]:
uniq2 = df['Country Name'].unique()
print("There are "+ str(len(uniq2)) + " countries in our data: ")
print(uniq2)

There are 103 countries in our data: 
['Albania' 'United Arab Emirates' 'Argentina' 'Australia' 'Austria'
 'Belgium' 'Burkina Faso' 'Bangladesh' 'Bulgaria' 'Bahrain' 'Bolivia'
 'Brazil' 'Botswana' 'Canada' 'Switzerland' 'Chile' 'China' 'Colombia'
 'Costa Rica' 'Cyprus' 'Czech Republic' 'Germany' 'Denmark'
 'Dominican Republic' 'Algeria' 'Ecuador' 'Egypt, Arab Rep.' 'Spain'
 'Estonia' 'Ethiopia' 'Finland' 'France' 'United Kingdom' 'Georgia'
 'Ghana' 'Greece' 'Guatemala' 'Honduras' 'Croatia' 'Hungary' 'Indonesia'
 'India' 'Ireland' 'Iceland' 'Israel' 'Italy' 'Jamaica' 'Jordan' 'Japan'
 'Kazakhstan' 'Kenya' 'Kyrgyz Republic' 'Cambodia' 'Kuwait' 'Sri Lanka'
 'Lesotho' 'Lithuania' 'Luxembourg' 'Latvia' 'Morocco' 'Moldova'
 'Madagascar' 'Mexico' 'Mali' 'Malta' 'Mongolia' 'Mauritania' 'Mauritius'
 'Malawi' 'Malaysia' 'Namibia' 'Nigeria' 'Nicaragua' 'Netherlands'
 'Norway' 'Nepal' 'New Zealand' 'Pakistan' 'Panama' 'Peru' 'Philippines'
 'Poland' 'Portugal' 'Paraguay' 'Romania' 'Russian Federati

In [34]:
grouped_df = df1.groupby("Indicator")
grouped_df = grouped_df.agg({"Country Name": "nunique"})
grouped_df = grouped_df.reset_index()
grouped_df = grouped_df.rename(columns={'Country Name': 'Country Count'})
grouped_df = grouped_df.sort_values('Country Count')
grouped_df

Indicator  Country Count
0  Global Gender Gap Economic Participation and O...            103
1  Global Gender Gap Educational Attainment Subindex            103
2     Global Gender Gap Health and Survival Subindex            103
3   Global Gender Gap Political Empowerment subindex            103
4                    Overall Global Gender Gap Index            103

From the above table, we know that each indicator contains 103 country data.